snapcraft/main.py

```diff
@@ -19,82 +19,87 @@
 snapcraft
 
 Usage:
-  snapcraft [--version | --help] [options] COMMAND [ARGS ...]
   snapcraft [options]
+  snapcraft [options] init
+  snapcraft [options] pull [<part> ...]
+  snapcraft [options] build [<part> ...]
+  snapcraft [options] stage [<part> ...]
+  snapcraft [options] strip [<part> ...]
+  snapcraft [options] clean [<part> ...] [--step <step>]
+  snapcraft [options] snap [<directory> --output <snap-file>]
+  snapcraft [options] cleanbuild
+  snapcraft [options] login
+  snapcraft [options] logout
+  snapcraft [options] upload <snap-file>
+  snapcraft [options] list-plugins
+  snapcraft [options] help (topics | <plugin> | <topic>) [--devel]
+  snapcraft (-h | --help)
+  snapcraft --version
 
 Options:
-  -h --help              show this help message and exit
-  -v --version           show program version and exit
-  -d --debug             print debug information while executing (including
-                         backtraces)
-  --no-parallel-build    use only a single build job per part (the default
-                         number of jobs per part is equal to the number of
-                         CPUs)
-  --target-arch ARCH     EXPERIMENTAL: sets the target architecture. Very few
-                         plugins support this.
+  -h --help                             show this help message and exit
+  -v --version                          show program version and exit
+  -d --debug                            print debug information while executing
+                                        (including backtraces)
+  --no-parallel-build                   use only a single build job per part
+                                        (the default number of jobs per part is
+                                        equal to the number of CPUs)
+  --target-arch ARCH                    EXPERIMENTAL: sets the target
+                                        architecture. Very few plugins support
+                                        this.
+  -s <step>, --step <step>              only clean the specified step and those
+                                        that depend upon it. <step> can be one
+                                        of: pull, build, stage or strip.
```

Comment:
It doesn't really matter. I can break down the header (Options) into "Options for cleaning", "Options for building"
and then just Options.

*Meaning of [options] shortcut slightly changed. Previously it meant "any known option". Now it means "any option not in usage-pattern". This avoids the situation when an option is allowed to be repeated unintentionally.*

from the first lines of https://github.com/docopt/docopt